blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServerBuilder.scala

```diff
@@ -309,88 +359,57 @@ class BlazeServerBuilder[F[_]](
   }
 
   def resource: Resource[F, Server[F]] = tickWheelResource.flatMap { scheduler =>
-    def resolveAddress(address: InetSocketAddress) =
-      if (address.isUnresolved) new InetSocketAddress(address.getHostName, address.getPort)
-      else address
-
-    val mkFactory: Resource[F, ServerChannelGroup] = Resource.make(F.delay {
-      if (isNio2)
-        NIO2SocketServerGroup
-          .fixedGroup(connectorPoolSize, bufferSize, channelOptions, selectorThreadFactory)
-      else
-        NIO1SocketServerGroup
-          .fixedGroup(connectorPoolSize, bufferSize, channelOptions, selectorThreadFactory)
-    })(factory => F.delay { factory.closeGroup() })
-
-    def mkServerChannel(factory: ServerChannelGroup): Resource[F, ServerChannel] =
-      Resource.make(F.delay {
-        val address = resolveAddress(socketAddress)
-
-        // if we have a Failure, it will be caught by the effect
-        factory.bind(address, pipelineFactory(scheduler)).get
-      })(serverChannel => F.delay { serverChannel.close() })
-
-    def logStart(server: Server[F]): Resource[F, Unit] =
-      Resource.liftF(F.delay {
-        Option(banner)
-          .filter(_.nonEmpty)
-          .map(_.mkString("\n", "\n", ""))
-          .foreach(logger.info(_))
-
-        logger.info(
-          s"http4s v${BuildInfo.version} on blaze v${BlazeBuildInfo.version} started at ${server.baseUri}")
-      })
-
-    Resource.liftF(verifyTimeoutRelations()) >>
-      mkFactory
-        .flatMap(mkServerChannel)
-        .map[Server[F]] { serverChannel =>
-          new Server[F] {
-            val address: InetSocketAddress =
-              serverChannel.socketAddress
-
-            val isSecure = sslBits.isDefined
-
-            override def toString: String =
-              s"BlazeServer($address)"
+    Resource.liftF(sslContext).flatMap { sslCtxOpt =>
+      def resolveAddress(address: InetSocketAddress) =
+        if (address.isUnresolved) new InetSocketAddress(address.getHostName, address.getPort)
+        else address
+
+      val mkFactory: Resource[F, ServerChannelGroup] = Resource.make(F.delay {
+        if (isNio2)
+          NIO2SocketServerGroup
+            .fixedGroup(connectorPoolSize, bufferSize, channelOptions, selectorThreadFactory)
+        else
+          NIO1SocketServerGroup
+            .fixedGroup(connectorPoolSize, bufferSize, channelOptions, selectorThreadFactory)
+      })(factory => F.delay { factory.closeGroup() })
+
+      def mkServerChannel(factory: ServerChannelGroup): Resource[F, ServerChannel] =
+        Resource.make(F.delay {
+          val address = resolveAddress(socketAddress)
+
+          val sslConfig = sslCtxOpt.map(ctx => (ctx, toSSLParameters(tlsParameters)))
```

Comment:
A troublesome point with this design is that it completely clobbers the default parameters from the `SSLContext`.  We should probably make `tlsParameters` an `Option`